### PR TITLE
Add description of the new endpoint of the config service [ECR-1436]

### PR DESCRIPTION
**Pull request by SerhiiOryshych**

### CODE REVIEW

File: src/advanced/configuration-updater.md [hunk 1]
```diff
@@ -100,10 +101,16 @@ config][stored_configuration] serialization. It has the following fields:
...
-- **validators**: Array<PublicKey\>  
+- **validator_keys**: Array<PublicKey\>  
   List of validators' public keys.
+- **majority_count**: integer  
+  Amount of votes required to commit a new configuration proposal.
+  By default the number of votes is calculated as
+  2/3 + 1 of total validators count.
 - **consensus**: Object  
   Consensus-specific configuration parameters.
+- **consensus.max_message_len**: integer
```
Comment (slowli) [inline src/advanced/configuration-updater.md]: For future-proofness, I suggest to remove all `consensus.*` sub-entries from here and instead link the `consensus` entry to the "[genesis.consensus]" section in the Configuration article.

File: src/advanced/configuration-updater.md [hunk 2]
```diff
@@ -418,7 +459,29 @@ and its signature are computed automatically.
 JSON object with the following fields:
 
 - **tx_hash**: Hash  
-  Hash of the corresponding `TxConfigVote` transaction.
+  Hash of the corresponding `Vote` transaction.
+
+### Submit Vote against Proposal
+
+```none
+POST {base_path}/configs/{config_hash_vote_for}/postagainst
```
Comment (slowli) [inline src/advanced/configuration-updater.md]: Nitpick: should be `{config_hash_vote_against}`.

File: src/advanced/configuration-updater.md [hunk 3]
```diff
@@ -333,11 +340,11 @@ with a state change if all of the following conditions take place:
   to the same configuration hash
 
 If all the checks pass, the execution results in modifying the `config_proposes`
```
Comment (slowli) [inline src/advanced/configuration-updater.md]: Off-topic: IMO, this explanation is unhelpful. I'd suggest to replace it with something like

> If all the checks pass, the proposal is recorded as a candidate for the next configuration. The validators can then vote for or against the proposal.

Same for the corresponding descriptions for `Vote` and `VoteAgainst` transactions.

File: src/advanced/configuration-updater.md [hunk 4]
```diff
@@ -358,16 +346,50 @@ Vote transactions will only get submitted and executed with state change
...
+
+### Vote against Proposal
+
+`VoteAgainst` is a transaction that implements voting against a previously
+proposed configuration.
+
+#### Data Layout
+
+- **cfg_hash**: Hash  
+  Hash of configuration to vote against.
```
Comment (Maria-Nosyk) [inline src/advanced/configuration-updater.md]: Hash of the configuration to be voted against.

File: src/advanced/configuration-updater.md [hunk 5]
```diff
@@ -358,16 +346,50 @@ Vote transactions will only get submitted and executed with state change
...
+Signature of the transaction is verified against the public key
+specified in `from`.
+
+#### Execution
+
+Vote transactions will only get submitted and executed with state change
+if all of the following conditions take place:
+
+- `cfg_hash` references a known proposed configuration `cfg`
+- A following configuration isn’t present
```
Comment (Maria-Nosyk) [inline src/advanced/configuration-updater.md]: same as above

File: src/advanced/configuration-updater.md [hunk 6]
```diff
@@ -418,12 +440,34 @@ and its signature are computed automatically.
...
-  Hash of the corresponding `TxConfigVote` transaction.
+  Hash of the corresponding `Vote` transaction.
+
+### Submit Vote against Proposal
+
+```none
+POST {base_path}/configs/{config_hash_vote_against}/postagainst
+```
+
+Create a [`VoteAgainst` transaction](#configuration-proposal).
```
Comment (Maria-Nosyk) [inline src/advanced/configuration-updater.md]: Creates

File: src/advanced/configuration-updater.md [hunk 7]
```diff
@@ -418,12 +440,34 @@ and its signature are computed automatically.
...
+As with the previous endpoint, the `from` field of the transaction
+and its signature are computed automatically.
+
+#### Parameters
+
+- **config_hash_vote_against**: Hash  
+  Hash of the configuration to vote against.
```
Comment (Maria-Nosyk) [inline src/advanced/configuration-updater.md]: to be voted against.

File: src/architecture/configuration.md [hunk 8]
```diff
@@ -68,21 +68,25 @@ two parts:
 
 [Consensus algorithm](consensus.md) parameters.
 
+- **max_message_len**  
+  Maximum message length (in bytes). This parameter determines the maximum
+  size of both consensus messages and transactions. The default value of the
+  parameter is 1 MB (1024 * 1024 bytes). The range of possible values for this
+  parameter is between 1MB and 2^32 bytes.
+- **max_propose_timeout**
+  Maximal propose timeout (in ms)
```
Comment (Maria-Nosyk) [inline src/architecture/configuration.md]: we use "Maximum" everywhere

File: src/architecture/configuration.md [hunk 9]
```diff
@@ -68,21 +68,25 @@ two parts:
...
+- **min_propose_timeout**
+  Minimal propose timeout (in ms)
```
Comment (Maria-Nosyk) [inline src/architecture/configuration.md]: Similarly "Minimum"

File: src/architecture/configuration.md [hunk 10]
```diff
@@ -68,21 +68,25 @@ two parts:
...
 - **peers_timeout**  
   Peers exchange timeout (in ms)
-- **timeout_adjuster**  
-  [Settings][ta-config] for the proposal timeout adjuster
+- **propose_timeout_threshold**
+  Amount of transactions in pool to start use `min_propose_timeout`
```
Comment (Maria-Nosyk) [inline src/architecture/configuration.md]: in the pool

**Pull request merged**